### PR TITLE
Add a 2nd time derivative auxkernel

### PR DESCRIPTION
**Pull request by GiudGiud**
I wanted to check on error messages for using 2nd order time derivatives with time integration schemes that dont implement them. It turns out it's safe, the time integrator does not request the 2nd time derivative and this brings an error.

but then I was mostly ready for this PR so just made it. 

refs #23770

### CODE REVIEW

File: framework/include/auxkernels/SecondTimeDerivativeAux.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+{
+public:
+  static InputParameters validParams();
+
+  SecondTimeDerivativeAux(const InputParameters & parameters);
+
+protected:
+  virtual Real computeValue() override;
+
+  /// Variable
```
Comment (lindsayad) [inline framework/include/auxkernels/SecondTimeDerivativeAux.h:27]:
```suggestion
  /// Variable we're computing the second time derivative of
```

File: framework/src/auxkernels/SecondTimeDerivativeAux.C [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+{
+  if (_use_qp_arg)
+  {
+    const auto qp_arg = std::make_tuple(_current_elem, _qp, _qrule);
+    return _factor(qp_arg) * _v[_qp];
+  }
+  else
+  {
+    const auto elem_arg = makeElemArg(_current_elem);
+    return _factor(elem_arg) * _v[_qp];
```
Comment (lindsayad) [inline framework/src/auxkernels/SecondTimeDerivativeAux.C:60]: I don't really like the mixing of the on-the-fly and pre-init'd data paradigms. It suggests we're missing capability one way or the other. Do you need one capability from one and one from another?

File: test/tests/auxkernels/time_derivative_second_aux/test.i [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    family = MONOMIAL
+    order = CONSTANT
+    fv = true
+  []
+[]
+
+[AuxKernels]
+  # Time derivative of a nonlinear variable
+  # Time derivative of an aux variable is not currently supported
+  # in the Auxiliary system (only in nonlinear)
```
Comment (lindsayad) [inline test/tests/auxkernels/time_derivative_second_aux/test.i:57]: really? I believe you, but I know we attach time integrators to the auxiliary system so conceptually I feel like we should be able to. Do you get an error?

File: test/tests/auxkernels/time_derivative_second_aux/tests [hunk 4]
```diff
@@ -0,0 +1,30 @@
+[Tests]
+  issues = "#442 #762"
+  design = "syntax/AuxKernels/index.md syntax/Kernels/index.md variables/AuxVariable.md syntax/Functions/index.md syntax/Materials/index.md"
+  [variable]
+    type = 'Exodiff'
+    input = 'test.i'
+    exodiff = 'test_out.e'
+    cli_args = "AuxVariables/inactive='variable_derivative_fv' AuxKernels/inactive='var_derivative_to_fv'"
+    requirement = "The system shall be able to compute the second time derivative of a non linear variable."
```
Comment (lindsayad) [inline test/tests/auxkernels/time_derivative_second_aux/tests:9]:
```suggestion
    requirement = "The system shall be able to compute the second time derivative of a nonlinear variable."
```

File: test/tests/auxkernels/time_derivative_second_aux/tests [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    ad_indexing_type = 'global'
```
Comment (lindsayad) [inline test/tests/auxkernels/time_derivative_second_aux/tests:10]: Ugh you have so many PRs that conflict with #23665 haha

File: test/tests/auxkernels/time_derivative_second_aux/tests [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  []
+
+  [errors]
+    requirement = 'The system shall error if'
+    [not_a_nodal_kernel]
+      type = RunException
+      input = 'test.i'
+      cli_args = 'AuxVariables/nodal/family=LAGRANGE AuxKernels/var_derivative/variable=nodal'
+      expect_err = 'This AuxKernel only supports Elemental fields'
+      detail = 'a second time derivative is requested but the variable is a nodal variable which is currently unsupported'
```
Comment (lindsayad) [inline test/tests/auxkernels/time_derivative_second_aux/tests:20]:
```suggestion
      detail = 'a second time derivative is requested but the variable is a nodal variable which is currently unsupported, and'
```

File: test/tests/auxkernels/time_derivative_second_aux/tests [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    []
+    [implicit_fe_fv_conversion]
+      type = RunException
+      input = 'test.i'
+      cli_args = "AuxVariables/inactive='' AuxKernels/inactive=''"
+      expect_err = "'v' argument is a finite element variable but 'variable' is not"
+      detail = 'a second time derivative for a finite element variable is to be saved in a finite volume variable'
```
Comment (lindsayad) [inline test/tests/auxkernels/time_derivative_second_aux/tests:27]:
```suggestion
      detail = 'a second time derivative for a finite element variable is requested to be stored in a finite volume variable.'
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: framework/src/auxkernels/SecondTimeDerivativeAux.C (hunk 2 above)
Comment (GiudGiud) [reply framework/src/auxkernels/SecondTimeDerivativeAux.C:60]: we are missing the dotdot in the functors. 
But is it really worth adding before we have a use for them?

File: test/tests/auxkernels/time_derivative_second_aux/test.i (hunk 3 above)
Comment (GiudGiud) [reply test/tests/auxkernels/time_derivative_second_aux/test.i:57]: this is pasted from the other test, where I wrote this most likely

I ll see what I can find

Comment (GiudGiud) [reply test/tests/auxkernels/time_derivative_second_aux/test.i]: ok seems to work. I tried two examples and got something reasonable with two integration schemes. 

I dont recall why I added this note

File: framework/src/auxkernels/SecondTimeDerivativeAux.C (hunk 2 above)
Comment (lindsayad) [reply framework/src/auxkernels/SecondTimeDerivativeAux.C:60]: I don't think it's worth it ... so my suggestion would be to not use functors for `_factor` ... but I suppose that `TimeDerivativeAux` uses functors for the factor and if `_factor` is really something people will use then it's annoying to support multiple parameter options, e.g. Function/CoupledVar/MaterialProperty ... so this is probably fine

**Review by lindsayad — APPROVED** [review]

**Review by lindsayad — CHANGES_REQUESTED** [review]
I'll submit a patch for `ad_indexing_type`

**Review by lindsayad — APPROVED** [review]

**Comment by GiudGiud** [comment]
Interesting that both u and the derivative differ. 
Seems like tightening u might be all that's needed

**Comment by GiudGiud** [comment]
fix is merged.

so the issue here is that the CD time integrator only supports the LINEAR solve_type, which has a very loose base tolerance (1e-5)

**Pull request merged**